libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/WebSocketEnabledHttpAdapter.cs

```diff
@@ -0,0 +1,179 @@
+        private readonly WebSocketConnector _webSocketConnector;
+        private readonly object initLock = new object();
+        private readonly List<Builder.IMiddleware> middlewares = new List<Builder.IMiddleware>();
+        private Lazy<bool> _ensureMiddlewareSet;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WebSocketEnabledHttpAdapter"/> class.
+        /// An adapter meant for subclassing and injecting custom implementation of BotFrameworkHttpAdapter for handling Http requests.
+        /// Throws <see cref="ArgumentNullException"/> if configure is null.
+        /// </summary>
+        /// <param name="configuration">The configuration for the adapter to use.</param>
+        /// <param name="credentialProvider">Optional credential provider.</param>
+        /// <param name="channelProvider">Optional channel provider.</param>
+        /// <param name="loggerFactory">Optional logger factory.</param>
+        public WebSocketEnabledHttpAdapter(IConfiguration configuration, ICredentialProvider credentialProvider = null, IChannelProvider channelProvider = null, ILoggerFactory loggerFactory = null)
+            : this(configuration, null, credentialProvider ?? new ConfigurationCredentialProvider(configuration), channelProvider ?? new ConfigurationChannelProvider(configuration), loggerFactory)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WebSocketEnabledHttpAdapter"/> class.
+        /// An adapter meant for subclassing and injecting custom implementation of BotFrameworkHttpAdapter for handling Http requests.
+        /// Throws <see cref="ArgumentNullException"/> if configure is null.
+        /// </summary>
+        /// <param name="configuration">The configuration for the adapter to use.</param>
+        /// <param name="botFrameworkHttpAdapter">Optional http adapter to use for non-streaming extensions requests.</param>
+        /// <param name="credentialProvider">Optional credential provider.</param>
+        /// <param name="channelProvider">Optional channel provider.</param>
+        /// <param name="loggerFactory">Optional logger factory.</param>
+        protected WebSocketEnabledHttpAdapter(IConfiguration configuration, IBotFrameworkHttpAdapter botFrameworkHttpAdapter = null, ICredentialProvider credentialProvider = null, IChannelProvider channelProvider = null, ILoggerFactory loggerFactory = null)
+        {
+            if (configuration == null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
+            }
+
+            var openIdEndpoint = configuration.GetSection(AuthenticationConstants.BotOpenIdMetadataKey)?.Value;
+
+            if (!string.IsNullOrEmpty(openIdEndpoint))
+            {
+                // Indicate which Cloud we are using, for example, Public or Sovereign.
+                ChannelValidation.OpenIdMetadataUrl = openIdEndpoint;
```

Comment:
This comment doesn't seem to match the code. A more detailed "If an open Id endpiont if found, use it. This enables soverign or public clouds..."